app/src/main/res/layout/fragment_view_image.xml

```diff
@@ -24,18 +24,54 @@
         app:layout_constraintRight_toRightOf="parent"
         app:layout_constraintTop_toTopOf="parent" />
 
-    <TextView
-        android:id="@+id/mediaDescription"
+    <!--    This should be inside CoordinatorLayout for two reasons:
+
+            1. TouchImageView really wants some constraints ans has no size otherwise
+            2. We don't want sheet to overlap with appbar but the only way to do it with autosizing
+            is to gibe parent some margin. -->
+    <androidx.coordinatorlayout.widget.CoordinatorLayout
         android:layout_width="match_parent"
-        android:layout_height="wrap_content"
-        android:background="#60000000"
-        android:hyphenationFrequency="full"
-        android:lineSpacingMultiplier="1.1"
-        android:padding="8dp"
-        android:textAlignment="center"
-        android:textColor="#eee"
-        android:textSize="?attr/status_text_medium"
+        android:layout_height="0dp"
+        android:layout_marginTop="70dp"
         app:layout_constraintBottom_toBottomOf="parent"
-        tools:text="Some media description" />
+        app:layout_constraintLeft_toLeftOf="parent"
+        app:layout_constraintRight_toRightOf="parent"
+        app:layout_constraintTop_toTopOf="parent">
+
+        <LinearLayout
+            android:id="@+id/captionSheet"
+            android:layout_width="match_parent"
+            android:layout_height="wrap_content"
+            android:background="@drawable/description_bg_expanded"
+            android:orientation="vertical"
+            app:behavior_peekHeight="90dp"
+            app:layout_behavior="com.google.android.material.bottomsheet.BottomSheetBehavior">
+
+            <ImageView
```

Comment:
Better make that just a `View` and set the handle drawable as background

Reply:
This produces a colored square for some reason, not sure why it decides to stretch

Reply:
Because of `android:layout_height="24dp"`probably. Try something lower, maybe adjust margin accordingly